app/serializers/activitypub/encrypted_message_serializer.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class ActivityPub::EncryptedMessageSerializer < ActivityPub::Serializer
+  context_extensions :olm
+
+  class DeviceSerializer < ActivityPub::Serializer
+    attributes :type, :device_id
+
+    def type
+      'Device'
+    end
+
+    def device_id
+      object
+    end
+  end
+
+  attributes :type, :message_type, :cipher_text, :message_franking
```

Comment:
My talk about having the message_franking being arbitrary data means the digest (`FrankingDigest`?) needs to be moved to the `EncryptedMessage`. In case of multiple message frankings (local and upstream servers), the digest would be the same for both anyway.